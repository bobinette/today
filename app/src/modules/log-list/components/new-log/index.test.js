import React from 'react';
import { fromJS } from 'immutable';

import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { NewLogInput } from '.';

configure({ adapter: new Adapter() });

test('render the log list', () => {
  const props = {
    actions: {
      createLog: jest.fn(),
    },
  };

  const tree = shallow(<NewLogInput {...props} />);
  expect(tree).toMatchSnapshot();
});

test('the textarea should appear when focused', () => {
  const props = {
    actions: { createLog: jest.fn() },
  };

  const tree = shallow(<NewLogInput {...props} />);
  tree.setState({ hasFocus: true });
  expect(tree.find('.NewLogInput__Content')).toMatchSnapshot();
});
