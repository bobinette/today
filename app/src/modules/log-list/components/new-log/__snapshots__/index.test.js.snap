// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the log list 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewLogInput
    actions={
      Object {
        "createLog": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="NewLogInput__Title"
          onChange={[Function]}
          placeholder="You don't know what happened today..."
        />,
        false,
      ],
      "className": "NewLogInput",
      "onBlur": [Function],
      "onFocus": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "NewLogInput__Title",
          "onChange": [Function],
          "placeholder": "You don't know what happened today...",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="NewLogInput__Title"
            onChange={[Function]}
            placeholder="You don't know what happened today..."
          />,
          false,
        ],
        "className": "NewLogInput",
        "onBlur": [Function],
        "onFocus": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "NewLogInput__Title",
            "onChange": [Function],
            "placeholder": "You don't know what happened today...",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`the textarea should appear when focused 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): ShallowWrapper {
    "length": 1,
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <NewLogInput
      actions={
        Object {
          "createLog": [MockFunction],
        }
      }
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="NewLogInput__Title"
            onChange={[Function]}
            placeholder="You don't know what happened today..."
          />,
          <div>
            <TextareaAutosize
              className="NewLogInput__Content"
              onChange={[Function]}
              onHeightChange={[Function]}
              placeholder="Tell me all about your day..."
              useCacheForDOMMeasurements={false}
              value=""
            />
            <div
              className="flex flex-align-right"
            >
              <button
                className="btn btn-primary btn-sm"
                onClick={[Function]}
              >
                Create
              </button>
            </div>
          </div>,
        ],
        "className": "NewLogInput NewLogInput--focus",
        "onBlur": [Function],
        "onFocus": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "NewLogInput__Title",
            "onChange": [Function],
            "placeholder": "You don't know what happened today...",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextareaAutosize
                className="NewLogInput__Content"
                onChange={[Function]}
                onHeightChange={[Function]}
                placeholder="Tell me all about your day..."
                useCacheForDOMMeasurements={false}
                value=""
              />,
              <div
                className="flex flex-align-right"
              >
                <button
                  className="btn btn-primary btn-sm"
                  onClick={[Function]}
                >
                  Create
                </button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "NewLogInput__Content",
                "onChange": [Function],
                "onHeightChange": [Function],
                "placeholder": "Tell me all about your day...",
                "useCacheForDOMMeasurements": false,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-primary btn-sm"
                  onClick={[Function]}
                >
                  Create
                </button>,
                "className": "flex flex-align-right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Create",
                  "className": "btn btn-primary btn-sm",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Create",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="NewLogInput__Title"
              onChange={[Function]}
              placeholder="You don't know what happened today..."
            />,
            <div>
              <TextareaAutosize
                className="NewLogInput__Content"
                onChange={[Function]}
                onHeightChange={[Function]}
                placeholder="Tell me all about your day..."
                useCacheForDOMMeasurements={false}
                value=""
              />
              <div
                className="flex flex-align-right"
              >
                <button
                  className="btn btn-primary btn-sm"
                  onClick={[Function]}
                >
                  Create
                </button>
              </div>
            </div>,
          ],
          "className": "NewLogInput NewLogInput--focus",
          "onBlur": [Function],
          "onFocus": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "NewLogInput__Title",
              "onChange": [Function],
              "placeholder": "You don't know what happened today...",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TextareaAutosize
                  className="NewLogInput__Content"
                  onChange={[Function]}
                  onHeightChange={[Function]}
                  placeholder="Tell me all about your day..."
                  useCacheForDOMMeasurements={false}
                  value=""
                />,
                <div
                  className="flex flex-align-right"
                >
                  <button
                    className="btn btn-primary btn-sm"
                    onClick={[Function]}
                  >
                    Create
                  </button>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "NewLogInput__Content",
                  "onChange": [Function],
                  "onHeightChange": [Function],
                  "placeholder": "Tell me all about your day...",
                  "useCacheForDOMMeasurements": false,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn btn-primary btn-sm"
                    onClick={[Function]}
                  >
                    Create
                  </button>,
                  "className": "flex flex-align-right",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Create",
                    "className": "btn btn-primary btn-sm",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Create",
                  "type": "button",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "NewLogInput__Content",
      "onChange": [Function],
      "onHeightChange": [Function],
      "placeholder": "Tell me all about your day...",
      "useCacheForDOMMeasurements": false,
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "NewLogInput__Content",
        "onChange": [Function],
        "onHeightChange": [Function],
        "placeholder": "Tell me all about your day...",
        "useCacheForDOMMeasurements": false,
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
